prescient/simulator/config.py

```diff
@@ -74,334 +413,93 @@ def outer_parse(args=None, values=None):
-        domain=bool,
-        default=False,
-        description="Allows quick start generators to be committed if load shedding occurs",
-    )).declare_as_argument()
-
-    simulator_config.declare("day_ahead_pricing", ConfigValue(
-        domain=In(["LMP", "EMLP", "aCHP"]),
-        default="aCHP",
-        description="Choose the pricing mechanism for the day-ahead market. Choices are "
-                    "LMP -- locational marginal price, "
-                    "ELMP -- enhanced locational marginal price, and "
-                    "aCHP -- approximated convex hull price. "
-                    "Default is aCHP.",
-    )).declare_as_argument()
-
-    # # PRESCIENT_OUTPUT_OPTIONS
-
-    simulator_config.declare("output_ruc_initial_conditions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output ruc (deterministic or stochastic) initial conditions prior "
-                    "to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_ruc_solutions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output ruc solutions following each solve."
-                    " Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_sced_initial_conditions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output sced initial conditions prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_sced_demands", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output sced demands prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_solver_logs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output solver logs during execution.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_max_decimal_places", ConfigValue(
-        domain=PositiveInt,
-        default=6,
-        description="When writing summary files, this rounds the output to the "
-                    "specified accuracy. Default is 6.",
-    )).declare_as_argument()
-
-    simulator_config.declare("disable_stackgraphs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Disable stackgraph generation",
-    )).declare_as_argument()
-
-    return simulator_config
-
-PrescientConfig = create_Prescient_Config()
-
-def clear_prescient_config():
-    global PrescientConfig
-    PrescientConfig = create_Prescient_Config()
+class _PluginPath(Path):
+    ''' A Path that registers a plugin when its path is set
+    '''
+    def __init__(self, parent_config: PrescientConfig):
+        self.config = parent_config
+        super().__init__()
+
+    def __call__(self, data):
+        path = super().__call__(data)
+        self.config.plugin_context.register_plugin(path, self.config)
```

Comment:
Is it a problem that the plugin is registered each time it is set?  What happens if one source sets a plugin, and another overrides that entry to something else?  I don't see a way for the overridden option to "unregister" that plugin?

Reply:
As this PR is it would restrict users to exactly one source for specifying options: (1) calling Prescient from the command line, (2) using a Prescient `runner.py` text file as the only option to `Prescient.simulate`, or (3) specifying the options as keyword arguments to `Prescient.simulate`.

In order to take full advantage of the Pyomo config system's flexibility we would need to sort this out.